### PR TITLE
[TFLITE]Select op support for tflite frontend

### PR DESCRIPTION
**Pull request by siju-samuel**
Added the support of `select` op in tflite frontend.

@FrozenGene @masahi please help me to review and merge this PR. TIA
Note: tflite `where` op is same as `select` in tflite.

Some part of code taken from #5474

### CODE REVIEW

File: python/tvm/relay/frontend/tflite.py [hunk 1]
```diff
@@ -2357,6 +2371,20 @@ def get_expr(self, input_tensor_idx):
     def has_expr(self, input_tensor_idx):
         return self.exp_tab.has_expr(get_tensor_name(self.subgraph, input_tensor_idx))
 
+    def get_tensor_or_const_expr(self, tensor):
```
Comment (FrozenGene) [inline python/tvm/relay/frontend/tflite.py:2374]: This name is trick. Here, we want to get relay expr and distinguish const. However, const is also expr in fact. Maybe we could name it `get_tensor_expr`? Do you have any better suggestion?

File: python/tvm/relay/frontend/tflite.py [hunk 2]
```diff
@@ -109,16 +109,17 @@ def __init__(self, model, subgraph, exp_tab):
...
             'PRELU': self.convert_prelu,
-            'REDUCE_ANY': self._convert_reduce_any,
-            'REDUCE_MAX': self._convert_reduce_max,
-            'REDUCE_MIN': self._convert_reduce_min,
-            'REDUCE_PROD': self._convert_reduce_prod,
+            'REDUCE_ANY': self.convert_reduce_any,
+            'REDUCE_MAX': self.convert_reduce_max,
+            'REDUCE_MIN': self.convert_reduce_min,
+            'REDUCE_PROD': self.convert_reduce_prod,
             'RELU':self.convert_relu,
```
Comment (u99127) [inline python/tvm/relay/frontend/tflite.py:116]: Why are these changes relevant to this Pull request ?

Comment (siju-samuel) [reply python/tvm/relay/frontend/tflite.py:116]: @u99127 for code consistency i changed, its not related to this PR. 
This 5 methods were starting with `_` and i removed those.

Comment (u99127) [reply python/tvm/relay/frontend/tflite.py:116]: In an ideal world that would be a separate PR rather than merging it in here as that could just go in separately and pretty mechanically.

However, that's really not our development practice yet. 

Ramana

Comment (siju-samuel) [reply python/tvm/relay/frontend/tflite.py]: @u99127 Thanks for the review. I totally agree with you. I removed those changes as part of this PR. i will raise another PR[#5515] for those. Could you please check again. TIA.

**Comment by siju-samuel** [comment]
@FrozenGene could you please help to merge this PR. TIA

**Comment by FrozenGene** [comment]
LGTM. @siju-samuel After thinking about this pr, I think we should make `get_expr` and `exp_tab.new_const` be private and only public `get_tensor_expr`. Other places use `get_expr` / `exp_tab.new_const` should be replaced as `get_tensor_expr`. Wish you could consider this suggestion after this pr.

**Comment by FrozenGene** [comment]
Thanks @siju-samuel @u99127 merged now.

**Comment by siju-samuel** [comment]
> LGTM. @siju-samuel After thinking about this pr, I think we should make get_expr and exp_tab.new_const be private and only public get_tensor_expr. Other places use get_expr / exp_tab.new_const should be replaced as get_tensor_expr. Wish you could consider this suggestion after this pr.

@FrozenGene Thanks. I will optimize all this and will do a PR.

**Comment by maheshambule** [comment]
> LGTM. @siju-samuel After thinking about this pr, I think we should make `get_expr` and `exp_tab.new_const` be private and only public `get_tensor_expr`. Other places use `get_expr` / `exp_tab.new_const` should be replaced as `get_tensor_expr`. Wish you could consider this suggestion after this pr.

@FrozenGene, @siju-samuel, Considering this fix as part of this refactoring https://github.com/apache/incubator-tvm/pull/5528.

Please let us know your opinions on other refactoring aspects as well.

**Comment by siju-samuel** [comment]
@maheshambule Thanks. 
You can consider it as part of #5528. .
